test/testInnosentStateMachine.js

```diff
@@ -0,0 +1,275 @@
+const chai = require('chai')
+const { describe, it } = require('mocha')
+const moment = require('moment')
+
+const expect = chai.expect
+const STATE = require('../SessionStateEnum.js')
+const OD_FLAG_STATE = require('../SessionStateODFlagEnum.js')
+const DOOR_STATE = require('../SessionStateDoorEnum.js')
+const StateMachine = require('../InnosentStateMachine.js')
+
+function setupDB(location_data = {}, session = {}, is_overdose_suspected = false) {
+  return {
+    getLocationData() {
+      return location_data
+    },
+    getMostRecentSession() {
+      return session
+    },
+    isOverdoseSuspectedInnosent() {
+      return is_overdose_suspected
+    },
+  }
+}
+
+function setupRedis(states = {}, door = {}, innosentHistory = [0], innosent = {}) {
+  return {
+    getLatestLocationStatesData() {
+      // initial state
+      return states
+    },
+    getLatestInnosentSensorData() {
+      return innosent
+    },
+    getLatestDoorSensorData() {
+      return door
+    },
+    addStateMachineData() {},
+    getInnosentWindow() {
+      return innosentHistory
+    },
+  }
```

Comment:
Hahaha, old school. I know you just copied this from the old state machine tests, but I find it amusing because these were essentially mocks before we added the `sinon` library. Not worth changing, but funny to see.